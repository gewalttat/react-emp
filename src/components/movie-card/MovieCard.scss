.movie-year {
  margin-left: 150px;
  padding-left: 10px;
  width: 66px;
  height: 26px;
  mix-blend-mode: normal;
  opacity: 0.5;
  border: 1px solid #979797;
  box-sizing: border-box;
  border-radius: 4px;
}

.card-menu {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 36px;
  height: 36px;
  background-color: #2a202d;
  position: absolute;
  margin-top: 20px;
  margin-left: 270px;
  z-index: 20;
  border-radius: 50%;

  & svg {
    fill: white;
  }
}
