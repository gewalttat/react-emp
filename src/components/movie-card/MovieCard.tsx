import React, { FC } from 'react';

export const MovieCard: FC = () => {
    return (
        <div className='movie-card'>
            <div className='movie-card-poster'>
                <div className='circle-button'></div>
            </div>
            <div className='movie-name'></div>
            <div className='movie-genre'></div>
            <div className='movie-year'></div>
        </div>
    )
}