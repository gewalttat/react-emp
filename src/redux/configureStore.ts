/* eslint-disable @typescript-eslint/ban-ts-comment */
import rootReducer, { rootSaga } from './rootReducer';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import createSagaMiddleware, { END } from 'redux-saga';

const sagaMiddleware = createSagaMiddleware();

const createStore = () => {
  const store = configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(sagaMiddleware),
  });

  // @ts-ignore
  store.runSaga = () => sagaMiddleware.run(rootSaga);
  // @ts-ignore
  store.close = () => store.dispatch(END);

  return store;
};

export default createStore;