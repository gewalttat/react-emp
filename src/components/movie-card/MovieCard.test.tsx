import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MovieCard } from './MovieCard';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

describe('With React Testing Library', () => {
    const props = {
        movieData: {
            budget: 55000000,
            genres: [''],
            id: 337167,
            overview: "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
            poster_path: "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
            release_date: "2018-02-07",
            revenue: 136906000,
            runtime: 106,
            tagline: "Don't miss the climax",
            title: "Fifty Shades Freed",
            vote_average: 6.1,
            vote_count: 1195
        }
    }

    const initialState = { output: 10 }
    const mockStore = configureStore();

    it('shows movie card', () => {
        const store = mockStore(initialState);
        const { getByText } = render(<Provider store={store}><MovieCard movieData={props.movieData} /></Provider>);
        expect(getByText("Fifty Shades Freed")).not.toBeNull();
    });

    it('renders correctly', () => {
        const store = mockStore(initialState);
        const tree = renderer.create(<Provider store={store}><MovieCard movieData={props.movieData} /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
      });
});