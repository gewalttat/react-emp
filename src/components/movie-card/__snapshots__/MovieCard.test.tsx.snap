// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library renders correctly 1`] = `
<div
  onMouseEnter={[Function]}
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-82opd5-MuiPaper-root-MuiCard-root"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <img
        alt="card image"
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        height="486"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
      />
      <div
        className="MuiCardContent-root css-1qu37se-MuiCardContent-root"
      >
        <div
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-46tk9y-MuiTypography-root"
        >
          Fifty Shades Freed
          <div
            className="movie-year"
          >
            2018
          </div>
        </div>
        <p
          className="MuiTypography-root MuiTypography-body2 css-je32x7-MuiTypography-root"
        >
          
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</div>
`;
